src/future.cpp

```diff
@@ -183,11 +198,58 @@ struct timed_queue {
     }
 
 };
-
 #endif
 
 /**************************************************************************************************/
 
+#elif STLAB_TASK_SYSTEM == STLAB_TASK_SYSTEM_WINDOWS
+
+class task_system
+{
+    PTP_POOL            _thread_pool;
+    TP_CALLBACK_ENVIRON _callback_environment;
+
+public:
+
+    task_system() {
+        InitializeThreadpoolEnvironment(&_callback_environment);
+        _thread_pool = CreateThreadpool(NULL);
```

Comment:
Instead of creating an additional thread pool it would be better to use the default pool, to do that you simply pass nullptr as the callback environment for CreateThreadpoolWork.